### PR TITLE
Tier manage GUI initial implementation.

### PR DESCRIPTION
**Pull request by chamindias**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIDescriptionGenUtil.java [hunk 1]
```diff
@@ -72,6 +75,76 @@ public static String generateDescriptionFromPolicy(OMElement policy) throws APIM
...
+            //converting to seconds
+            timeDurationSeconds = TimeUnit.MILLISECONDS.toSeconds(Long.parseLong(duration.getText().trim()));
+            return timeDurationSeconds;
+        } catch (OMException e) {
+            String errorMessage = "Policy could not be parsed correctly based on " + "http://schemas.xmlsoap.org/ws/2004/09/policy specification";
+            log.warn(errorMessage);
+            throw new APIManagementException(errorMessage);
+        } catch (NumberFormatException e) {
+            log.error("Error in retrieving request count in tier xml. " + e);
+            throw new APIManagementException("Error in retrieving request count in tier xml. " + e.getMessage());
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIDescriptionGenUtil.java]: log with exception log.error(msg, e)

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIDescriptionGenUtil.java [hunk 2]
```diff
@@ -72,6 +75,76 @@ public static String generateDescriptionFromPolicy(OMElement policy) throws APIM
...
+                        " element data not found empty in the policy.";
+                log.warn(message);
+                throw new APIManagementException(message);
+            }
+            requestCount = Long.parseLong(maxCount.getText().trim());
+            return requestCount;
+        } catch (OMException e) {
+            String errorMessage = "Policy could not be parsed correctly based on " +
+                    "http://schemas.xmlsoap.org/ws/2004/09/policy specification";
+            log.warn(errorMessage);
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIDescriptionGenUtil.java]: log with exception

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIDescriptionGenUtil.java (hunk 1 above)
Comment (chamindias) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIDescriptionGenUtil.java]: Fixed in PR : https://github.com/wso2/carbon-apimgt/pull/1311

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIDescriptionGenUtil.java (hunk 2 above)
Comment (chamindias) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIDescriptionGenUtil.java]: Fixed in PR : https://github.com/wso2/carbon-apimgt/pull/1311

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 3]
```diff
@@ -1521,6 +1525,54 @@ public static String createEndpoint(String endpointUrl, Registry registry) throw
...
+
+                    if (tierName != null && tierName.equalsIgnoreCase(id.getText())) {
+                        foundTier = true;
+                        policies.remove();
+                        break;
+                    }
+                }
+
+                if (!foundTier) {
+                    throw new APIManagementException("Tier doesn't exist!");
```
Comment (nuwand) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: The name of the Tier should be included in the log message.

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin-dashboard/site/pages/tier-edit.jag [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    var user = jagg.getUser();
+    if (!user) {
+        response.sendRedirect(getRedirectPathForNonAuthenticatedUser());
+        return;
+    }
+
+    jagg.render({
+        "name":"page/base",
+        "inputs":{
+            "title":"Tier Edit",
```
Comment (nuwand) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin-dashboard/site/pages/tier-edit.jag]: Title Should be "Edit Tiers", or even better "Update Tiers"

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java (hunk 3 above)
Comment (chamindias) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Done in : https://github.com/wso2/carbon-apimgt/pull/1382

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin-dashboard/site/pages/tier-edit.jag (hunk 4 above)
Comment (chamindias) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin-dashboard/site/pages/tier-edit.jag]: Done in : https://github.com/wso2/carbon-apimgt/pull/1382

**Pull request merged**